backend/src/main/java/me/magicbudget/controller/FriendController.java

```
@@ -0,0 +1,48 @@
+package me.magicbudget.controller;
+
+
+import me.magicbudget.dto.outgoingresponse.FriendResponse;
+import me.magicbudget.model.Friendship;
+import me.magicbudget.service.FriendshipService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import java.util.List;
+
+@RestController
+@RequestMapping("api/{userid}/friend")
+public class FriendController {
+
+  private final FriendshipService friendshipService;
+
+  public FriendController(FriendshipService friendshipService) {
+    this.friendshipService = friendshipService;
+  }
+
+  @PostMapping("/add-friend")
+  public ResponseEntity<String> addFriend(@PathVariable("userid") String userId, @RequestBody String username) {
+    try {
+      friendshipService.addFriend(userId, username);
+      return new ResponseEntity<>(HttpStatus.OK);
+    }
+    catch (Exception e) {
+      return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
+    }
+  }
+
+  @GetMapping("/")
+  public ResponseEntity<List<FriendResponse>> getFriends(@PathVariable("userid") String userId) {
+    try {
+      return new ResponseEntity<>(friendshipService.getFriends(userId),HttpStatus.OK);
+    }
+    catch (Exception e) {
```

Don't suppress errors